<launch>
  <arg name="name" />
  <arg name="device_path" />
  <arg name="camera_name" />
  <arg name="manager_name" />
  <arg name="resource_name" />
  <node pkg="nodelet" type="nodelet" name="$(arg name)"
        args="load v4l2_cam/camera_nodelet $(arg manager_name)" output="screen">
    <param name="camera_name" value="$(arg camera_name)" />
    <param name="device_path" value="$(arg device_path)" />
    <param name="resource_manager" value="$(arg resource_name)" />
    <rosparam>
      frame_resolution: [160, 120]
      frame_rate: 120
      stream_on_start: false
      
      resources:
        usb_bandwidth: 200
      
      controls:
        horizontal_flip:
          id: 9963796
          value: 0
        vertical_flip:
          id: 9963797
          value: 0
        sharpness: #0-63, (0 is no blur, 63 is max blur)
          id: 9963803
          value: 0
        auto_exposure:
          id: 10094849
          value: 0
        #exposure: #0-255
          #id: 9963793
          #value: 0
        power_line_freq:
          id: 9963800
          value: 0
        auto_gain:
          id: 9963795
          value: 1
        #gain: #0-63
          #id: 9963795
          #value: 0
        auto_white_balance:
          id: 9963788
          value: 1
        #brightness:
          #id: 993776
          #value: 0
        #contrast: #0-255
          #id: 9963777
          #value: 0
        #saturation: #0-255
          #id: 9963778
          #value: 0
        #hue: #-90-90
          #id: 9963779
          #value: 0
    </rosparam>
  </node>
</launch>
